pkg/app/pagewriter/sign_in.html

```diff
@@ -31,15 +31,20 @@
         {{.LogoData}}
       </div>
       {{ end }}
+      
+      {{ $name := .ProviderName }}
+      {{ range $index,$id   := .ProviderID }}
 
-      <form method="GET" action="{{.ProxyPrefix}}/start">
-        <input type="hidden" name="rd" value="{{.Redirect}}">
-          {{ if .SignInMessage }}
-          <p class="block">{{.SignInMessage}}</p>
+      <form method="GET" action="{{$.ProxyPrefix}}/{{$id}}/start">
```

Comment:
What happens if there are multiple providers and the ID isn't specified, eg, what happens when users hit the normal `/oauth2/start` path?

Have you considered sending the ID as a query string parameter rather than adding it into the path of the request?

Reply:
when normal path is chosen it defaults back to the [0]th provider - no functionality is lost.

i can add a parameter as an option too